application/helpers/export_helper.php

```diff
@@ -2834,11 +2834,15 @@ function tsvSurveyExport($surveyid)
         }
 
         // subquestions data
+        $subquestions_data = [];
         if (array_key_exists('subquestions', $xmlData)) {
+            if (isset($xmlData['subquestions']['rows']['row']['qid'])) {
+                // Only one subquestion then one row : set as array like we have multiple rows
```

Comment:
The question part (and maybe other) have a really more complex solution using a temporary array : maybe fix it too ?

Reply:
> Can you please add a link to the part code you refer to?

https://github.com/LimeSurvey/LimeSurvey/blob/a7ca24dd21fd009df31db9fc86b3e2a6c5fa166c/application/helpers/export_helper.php#L2811

Reply:
Default values 

https://github.com/LimeSurvey/LimeSurvey/blob/a7ca24dd21fd009df31db9fc86b3e2a6c5fa166c/application/helpers/export_helper.php#L2751-L2753
OK

groups
https://github.com/LimeSurvey/LimeSurvey/blob/a7ca24dd21fd009df31db9fc86b3e2a6c5fa166c/application/helpers/export_helper.php#L2783-L2788

Reply:
PS : i thibnk i prefer  array_key_exists('id', $myArray) VS ! array_key_exists(0, $myArray)